pkg/kubernetes/rollout/v1alpha1/rolloutstatus.go

```diff
@@ -0,0 +1,194 @@
+type rolloutOutput struct {
+	IsRolledout bool   `json:"IsRolledout"`
+	Message     string `json:"Message"`
+}
+
+// ExtnV1Beta1DeployRolloutStatusGenerator generates RolloutOutput from deploymnet object for that deployment
+var ExtnV1Beta1DeployRolloutStatusGenerator = func(deploy api_extn_v1beta1.Deployment) (op []byte, err error) {
+	r := rolloutOutput{}
+	r.IsRolledout = false
+	var cond *api_extn_v1beta1.DeploymentCondition
+	// list all conditions and and select that condition which type is Progressing.
+	for i := range deploy.Status.Conditions {
+		c := deploy.Status.Conditions[i]
+		if c.Type == api_extn_v1beta1.DeploymentProgressing {
+			cond = &c
+		}
+	}
+	// if deploy.Generation <= deploy.Status.ObservedGeneration then deployment spec is not updated yet.
+	// it marked IsRolledout as false and update message accordingly
+	if deploy.Generation <= deploy.Status.ObservedGeneration {
+		// If Progressing condition's reason is ProgressDeadlineExceeded then it is not rolled out.
+		if cond != nil && cond.Reason == "ProgressDeadlineExceeded" {
+			r.Message = fmt.Sprintf("Deployment exceeded its progress deadline")
+			return json.Marshal(r)
+		}
+		// if deploy.Status.UpdatedReplicas < *deploy.Spec.Replicas then some of the replicas are updated
+		// and some of them are not. It marked IsRolledout as false and update message accordingly
+		if deploy.Spec.Replicas != nil && deploy.Status.UpdatedReplicas < *deploy.Spec.Replicas {
+			r.Message = fmt.Sprintf("Waiting for deployment rollout to finish: %d out of %d new replicas have been updated",
+				deploy.Status.UpdatedReplicas, *deploy.Spec.Replicas)
+			return json.Marshal(r)
+		}
+		// if deploy.Status.Replicas > deploy.Status.UpdatedReplicas then some of the older replicas are in running state
+		// because newer replicas are not in running state. It waits for newer replica to come into reunning state then terminate.
+		// It marked IsRolledout as false and update message accordingly
+		if deploy.Status.Replicas > deploy.Status.UpdatedReplicas {
+			r.Message = fmt.Sprintf("Waiting for deployment rollout to finish: %d old replicas are pending termination",
+				deploy.Status.Replicas-deploy.Status.UpdatedReplicas)
+			return json.Marshal(r)
+		}
+		// if deploy.Status.AvailableReplicas < deploy.Status.UpdatedReplicas then all the replicas are updated but they are
+		// not in running state. It marked IsRolledout as false and update message accordingly.
+		if deploy.Status.AvailableReplicas < deploy.Status.UpdatedReplicas {
+			r.Message = fmt.Sprintf("Waiting for deployment rollout to finish: %d of %d updated replicas are available",
+				deploy.Status.AvailableReplicas, deploy.Status.UpdatedReplicas)
+		}
+		r.IsRolledout = true
+		r.Message = fmt.Sprintf("Deployment %q successfully rolled out", deploy.Name)
+		return json.Marshal(r)
+	}
+	r.Message = fmt.Sprintf("Waiting for deployment spec update to be observed")
+	return json.Marshal(r)
```

Comment:
I will come back with review comments, once you go ahead with above approach.

Reply:
Try implementing same for all the resources

Reply:
done